### DISCUSSION TITLE
Skin editor in "Excluding overlays" screen scaling mode can change playfield aspect ratio

### DISCUSSION THREAD

**Discussion by integerrr** [ppy/osu#14887] [Q&A]
## (Completely rewritten) Description
The gameplay area in skin editor, with `excluding overlays` screen scaling, uses the aspect ratio of the **monitor** instead of what's been set in screen scaling. This will not be an issue if the scaling chosen has the same aspect ratio as the player's monitor.

For example, I have a 16:10 monitor and in the video I had screen scaling set to `excluding overlays` with the scaling set to roughly 2:1 to exaggerate the effect and make it easier to see.

This usually wouldn't be an issue as normally UI elements hug the corners/edges and it doesnt make any difference if the gameplay aspect ratio is changed, but for certain "modern" skins which are designed for specific aspect ratios, there are UI elements (hp bars with a playfield border, accuracy counter that pads space to centralise the numbers etc.) that will look funky when gameplay aspect ratio changes entering the skin editor i.e. what you have in the editor wont match what you will see once exiting skin editor.

Resizing gameplay area in skin editor is ok as long as its aspect ratio doesn't change.

## Video showing the issue

https://user-images.githubusercontent.com/7279624/137999080-14412913-2196-4bf6-8d47-5cdd1414d565.mp4

**Comment by peppy** [comment]
I'm not sure what you're saying is wrong here. The skin editor intentionally resizes the gameplay area to take up the full available area, by design. Are you saying you don't want this to happen?

**Reply by integerrr** [reply]
Not the gameplay area, that is fine, what I meant in op are the other elements i.e. health bar, accuracy display, score display, UR bar etc.,

**Reply by peppy** [reply]
You're going to need to try and explain this from a different angle. I can't understand your original post.

**Reply by integerrr** [reply]
sorry for overcomplicating things in the op, i've rewritten the whole thing, also i realise i completely misunderstood your initial comment: yes the gameplay area can be resized as long as the aspect ratio remains

**Reply by peppy** [reply]
Right, so the case here is that you're designing a skin layout to only work for you at your custom aspect ratio? That you're not planning to share with others?

The editor was primarily designed with the flexibility in mind to allow sharing skins, but I guess I can understand the local user use case. Maybe this can be a toggle somewhere, will have to have a think about it.

**Reply by integerrr** [reply]
now that I think more of it, is it easier to just deprecate the use of elements like these? 
![scorebar-bg@2x](https://user-images.githubusercontent.com/7279624/138294338-87e71b38-c222-46b3-b74d-ba142dabb917.png)

Since these are the source of all evil in my op anyway, and now we can customize playfield size relative to the whole gameplay screen these are kind of useless in lazer.

**Reply by bdach** [reply]
Those sorts of elements were never actually supported. The fact that they ever worked in the first place is because they rely on stable undefined behaviour and implementation specifics.
